### PR TITLE
Fixed null and empty string date-picker values

### PR DESCRIPTION
**Pull request by TheOtherDude**
### This project uses [semver](semver.org), please check the scope of this pr:

- [ ] #none# - documentation fixes and/or test additions
- [x] #patch# - backwards-compatible bug fix
- [ ] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

* Passing `null` or `''` to `frost-date-picker` will now show as empty instead of invalid

### CODE REVIEW

File: addon/components/frost-date-picker.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ export default Component.extend(EventsProxyMixin, {
   @readOnly
   @computed('value')
   _value (value) {
-    if (value === undefined) {
+    if (!value) {
```
Comment (gknoy) [inline addon/components/frost-date-picker.js]: I like that this covers both `undefined` and `''`, but there are other falsey values (`0`) which probably we still want to mark as invalid.  I like this, but would be happier if it were more explicitly only for values that can represent emptiness (`undefined`, `''`).

**Comment by gknoy** [comment]
👍

File: addon/components/frost-date-picker.js (hunk 1 above)
Comment (TheOtherDude) [reply addon/components/frost-date-picker.js]: Hmmmmmmm, sure, I'll go with it

Comment (gknoy) [reply addon/components/frost-date-picker.js]: Thank you!

File: addon/components/frost-date-picker.js [hunk 2]
```diff
@@ -50,7 +50,8 @@ export default Component.extend(EventsProxyMixin, {
   @readOnly
   @computed('value')
   _value (value) {
-    if (value === undefined) {
+    const emptyValues = new Set(['', undefined, null])
+    if (value in emptyValues) {
```
Comment (juwara0) [inline addon/components/frost-date-picker.js]: Why not use `isEmpty()`. https://www.emberjs.com/api/ember/2.12/namespaces/Ember/methods/isEmpty?anchor=isEmpty

File: tests/integration/components/frost-date-picker-test.js [hunk 3]
```diff
@@ -99,6 +99,30 @@ describe(test.label, function () {
     })
   })
 
+  describe('when empty string is provided', function () {
```
Comment (juwara0) [inline tests/integration/components/frost-date-picker-test.js]: I do not see tests for `undefined` or `null`.

Comment (TheOtherDude) [reply tests/integration/components/frost-date-picker-test.js]: Undefined is tested directly above it. I didn't test null because I thought that was overkill. If you would like to see it, I will gladly add it for you.

Comment (juwara0) [reply tests/integration/components/frost-date-picker-test.js]: Yes, please add the test for `null`. The test above that is for `undefined` is incorrect. What we would want in that case is to not set the property at all so that we can validate that it is not being set to something else in that case.

File: addon/components/frost-date-picker.js (hunk 2 above)
Comment (TheOtherDude) [reply addon/components/frost-date-picker.js]: I originally was just checking for falsey values, but @gknoy asked to see the empty values explicitly. I'm fine with using `isEmpty()` if that is explicit enough for him.

**Comment by juwara0** [comment]
👍

**Comment by juwara0** [comment]
👍
